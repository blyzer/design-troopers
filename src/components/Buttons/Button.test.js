import React from 'react';
import {render, fireEvent, cleanup} from '@testing-library/react';

import { Button } from './index.js';

function setup(props = {}, config = {}) {
  const spyButtonClick = jest.fn();
  const { debug, getByRole } = render(
    <Button onClick={spyButtonClick} {...props}>Sign Up</Button>,
  );

  const triggerClick = (event) => fireEvent.click(getByRole('button'), event);

  return { debug, getByRole, triggerClick, spyButtonClick };
}

describe('<Button /> works', () => {
  test('it renders properly', () => {
    const { getByRole } = setup();
    expect(getByRole('button')).toMatchSnapshot();
  });

  test('on click button triggers callback', () => {
    const { triggerClick, spyButtonClick } = setup();
    triggerClick();
    triggerClick();
    expect(spyButtonClick).toHaveBeenCalledTimes(2);
  });
});
