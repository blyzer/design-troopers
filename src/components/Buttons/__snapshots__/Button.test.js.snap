// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> works it renders properly 1`] = `
<button
  class="dt-py-2 dt-px-6 dt-rounded dt-shadow-lg dt-border dt-border-gray-300 dt-flex dt-items-center dt-justify-center dt-transition-all dt-transition-500 focus:dt-outline-none focus:dt-shadow-outline"
  data-name="DTButton"
>
  Sign Up
</button>
`;
